docs/aws/audit/esmonitoring/rules/es_instance_counts_remediation.mdx

```diff
@@ -1,6 +1,256 @@
+3. Once you're on the Elasticsearch Service dashboard, you'll see a list of your Elasticsearch domains. Click on the name of the domain you want to check.
+4. On the domain dashboard, you can see the instance count under the "Elasticsearch cluster" section. It will show the number of instances currently running for that domain.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Elasticsearch domains: Use the following AWS CLI command to list all the Elasticsearch domains in your AWS account.
+
+   ```
+   aws es list-domain-names
+   ```
+
+   This command will return a list of all Elasticsearch domains in your AWS account.
+
+3. Describe the Elasticsearch domain: Once you have the list of all Elasticsearch domains, you can describe a specific domain to get more details about it, including the instance count. Use the following AWS CLI command to describe a specific Elasticsearch domain.
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name your_domain_name
+   ```
+
+   Replace 'your_domain_name' with the name of your Elasticsearch domain. This command will return a JSON output with all the details about the specified Elasticsearch domain.
+
+4. Check the instance count: In the JSON output returned by the previous command, look for the 'ElasticsearchClusterConfig' field. Under this field, you will find the 'InstanceCount' field which shows the number of instances in the specified Elasticsearch domain.
+
+   ```
+   "ElasticsearchClusterConfig": {
+       "InstanceType": "m3.medium.elasticsearch",
+       "InstanceCount": 1,
+       "DedicatedMasterEnabled": false,
+       "ZoneAwarenessEnabled": false,
+       "ZoneAwarenessConfig": {
+           "AvailabilityZoneCount": 2
+       }
+   }
+   ```
+
+   In this example, the 'InstanceCount' is 1, which means there is only one instance in this Elasticsearch domain.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: To interact with Elasticsearch in AWS, you need to install the `boto3` and `requests` libraries. You can install them using pip:
+
+```python
+pip install boto3 requests
+```
+
+2. Set up AWS credentials: Before you can interact with AWS services, you need to set up your AWS credentials. You can do this by setting the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables, or by using the `boto3` library's `Session` object:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>